src/MainWindow.vala

```diff
@@ -182,4 +183,51 @@ public class Maya.MainWindow : Hdy.ApplicationWindow {
 
         return base.configure_event (event);
     }
+
+    public override bool delete_event (Gdk.EventAny event) {
+        var loop = new MainLoop (MainContext.default (), true);
+        ((Application) application).ask_for_background.begin ((obj, res) => {
+            unowned var app = (Application) obj;
+            loop.quit ();
+
+            if (!app.ask_for_background.end (res)) {
+                app.release ();
+            }
+        });
+
+        if (loop.is_running ()) {
```

Comment:
Is this right?  It looks like you are running a loop that is already running?

Reply:
Yes, we use the loop to block the function while `application.ask_for_background` is running, so we should only call `loop.run()` if `is_running` is true.

The need to block the function here is that we don't delete the application window while the portal dialog is open.

Reply:
Updated this part to stop the event handling instead of waiting the method response to continue.